test/unit/utils/params.cpp

```diff
@@ -0,0 +1,240 @@
+        pUSMDesc = nullptr;
+        params.ppUSMDesc = &pUSMDesc;
+        size = 0;
+        params.psize = &size;
+    }
+
+    ur_usm_host_alloc_params_t *get_struct() { return &params; }
+};
+
+struct UrUsmHostAllocParamsEmpty : UrUsmHostAllocParams {
+    UrUsmHostAllocParamsEmpty() : UrUsmHostAllocParams() {}
+    const char *get_expected() {
+        return "\\.hContext = nullptr, \\.pUSMDesc = nullptr, \\.pool = "
+               "nullptr, "
+               "\\.size = 0, \\.ppMem = .+ \\(nullptr\\)";
+    };
+};
+
+template <>
+std::unique_ptr<WrappedParams<ur_usm_host_alloc_params_t>>
+createParams<UrUsmHostAllocParamsEmpty>() {
+    return std::make_unique<UrUsmHostAllocParamsEmpty>();
+}
+
+struct UrUsmHostAllocParamsUsmDesc : UrUsmHostAllocParams {
+    ur_usm_desc_t usm_desc;
+    UrUsmHostAllocParamsUsmDesc() : UrUsmHostAllocParams() {
+        usm_desc.align = 64;
+        usm_desc.flags = UR_USM_FLAG_BIAS_CACHED;
+        usm_desc.hints = UR_USM_ADVICE_SET_NON_ATOMIC_MOSTLY;
+        usm_desc.pNext = nullptr;
+        usm_desc.stype = UR_STRUCTURE_TYPE_USM_DESC;
+        pUSMDesc = &usm_desc;
+    }
+    const char *get_expected() {
+        return ".*\\.pUSMDesc = .+ \\(\\(struct "
+               "ur_usm_desc_t\\)\\{\\.stype = UR_STRUCTURE_TYPE_USM_DESC, "
+               "\\.pNext = "
+               "nullptr, \\.flags = UR_USM_FLAG_BIAS_CACHED, \\.hints = "
+               "UR_USM_ADVICE_SET_NON_ATOMIC_MOSTLY, \\.align = 64\\}\\).*";
+    };
+};
+
+template <>
+std::unique_ptr<WrappedParams<ur_usm_host_alloc_params_t>>
+createParams<UrUsmHostAllocParamsUsmDesc>() {
+    return std::make_unique<UrUsmHostAllocParamsUsmDesc>();
+}
+
+struct UrUsmHostAllocParamsHostDesc : UrUsmHostAllocParamsUsmDesc {
+    ur_usm_host_desc_t host_desc;
+    UrUsmHostAllocParamsHostDesc() : UrUsmHostAllocParamsUsmDesc() {
+        host_desc.flags = UR_USM_HOST_MEM_FLAG_INITIAL_PLACEMENT;
+        host_desc.pNext = nullptr;
+        host_desc.stype = UR_STRUCTURE_TYPE_USM_HOST_DESC;
+        usm_desc.pNext = &host_desc;
+    }
+    const char *get_expected() {
+        return ".*\\.pNext = .+ \\(\\(struct "
+               "ur_usm_host_desc_t\\)\\{\\.stype = "
+               "UR_STRUCTURE_TYPE_USM_HOST_DESC, "
+               "\\.pNext = "
+               "nullptr, \\.flags = "
+               "UR_USM_HOST_MEM_FLAG_INITIAL_PLACEMENT\\}\\).*";
+    };
+};
+
+template <>
+std::unique_ptr<WrappedParams<ur_usm_host_alloc_params_t>>
+createParams<UrUsmHostAllocParamsHostDesc>() {
+    return std::make_unique<UrUsmHostAllocParamsHostDesc>();
+}
+
+using testing::Types;
+typedef Types<UrInitParamsNoFlags, UrInitParamsInvalidFlags,
+              UrUsmHostAllocParamsEmpty, UrPlatformGetEmptyArray,
+              UrPlatformGetTwoPlatforms, UrUsmHostAllocParamsUsmDesc,
+              UrUsmHostAllocParamsHostDesc>
+    Implementations;
+
+using ::testing::MatchesRegex;
+using namespace ur_params;
+
+TYPED_TEST_SUITE(ParamsTest, Implementations, );
```

Comment:
nit: A comma at the end and a whitespace is here on purpose?

Reply:
yes, otherwise there's a warning about not enough macro arguments.